packages/odyssey-react-mui/src/useAutocomplete.tsx

```diff
@@ -0,0 +1,398 @@
+import { Field } from "./Field";
+import {
+  ComponentControlledState,
+  useInputValues,
+  getControlState,
+} from "./inputUtils";
+
+const ListboxContainer = styled.div({
+  width: "100%",
+  height: "100%",
+});
+
+// const AutoSizer = _AutoSizer as unknown as FC<AutoSizerProps>;
+type SetItemSize = (size: number) => void;
+
+export type UseAutocompleteProps<
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+> = Pick<
+  AutocompleteProps<OptionType, HasMultipleChoices, IsCustomValueAllowed>,
+  | "ariaDescribedBy"
+  | "defaultValue"
+  | "errorMessage"
+  | "errorMessageList"
+  | "hasMultipleChoices"
+  | "hint"
+  | "HintLinkComponent"
+  | "inputValue"
+  | "isFullWidth"
+  | "isOptional"
+  | "isVirtualized"
+  | "label"
+  | "name"
+  | "onChange"
+  | "onInputChange"
+  | "testId"
+  | "value"
+>;
+
+export const useAutocomplete = <
+  OptionType,
+  HasMultipleChoices extends boolean | undefined,
+  IsCustomValueAllowed extends boolean | undefined,
+>({
+  ariaDescribedBy,
+  defaultValue,
+  errorMessage,
+  errorMessageList,
+  hasMultipleChoices,
+  hint,
+  HintLinkComponent,
+  inputValue,
+  isFullWidth,
+  isOptional,
+  isVirtualized: isVirtualizedProp,
+  label,
+  name: nameOverride,
+  onChange: onChangeProp,
+  onInputChange: onInputChangeProp,
+  testId,
+  value,
+}: UseAutocompleteProps<
+  OptionType,
+  HasMultipleChoices,
+  IsCustomValueAllowed
+>) => {
+  const { t } = useTranslation();
+  const controlledStateRef = useRef(
+    getControlState({
+      controlledValue: value,
+      uncontrolledValue: defaultValue,
+    }),
+  );
+
+  const isVirtualized = useRef(Boolean(isVirtualizedProp));
+
+  const defaultValueProp = useMemo<
+    | AutocompleteValue<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >
+    | undefined
+  >(() => {
+    if (hasMultipleChoices) {
+      if (value === undefined) {
+        return defaultValue;
+      }
+      return [] as AutocompleteValue<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >;
+    }
+    return value === undefined ? defaultValue : undefined;
+  }, [defaultValue, hasMultipleChoices, value]);
+
+  const valueProps = useInputValues({
+    defaultValue: defaultValueProp,
+    value: value,
+    controlState: controlledStateRef.current,
+  });
+
+  const inputValueProp = useMemo(() => {
+    if (controlledStateRef.current === ComponentControlledState.CONTROLLED) {
+      return { inputValue };
+    }
+    return undefined;
+  }, [inputValue]);
+
+  const onChange = useCallback<
+    NonNullable<
+      MuiUseAutocompleteProps<
+        OptionType,
+        HasMultipleChoices,
+        undefined,
+        IsCustomValueAllowed
+      >["onChange"]
+    >
+  >(
+    (event, value, reason, details) => {
+      onChangeProp?.(event, value, reason, details);
+    },
```

Comment:
At this point, wouldn't it make sense to do `...args` instead?

```suggestion
    (...args) => {
      onChangeProp?.(...args);
    },
```

Reply:
I don't have strong feelings either way

Reply:
I'd like to do that then.